### DISCUSSION TITLE
how to acess the complete procedure call with all parameters inside the program

### DISCUSSION THREAD

**Discussion by brandy125** [gnudatalanguage/gdl#1902] [Q&A]
Is it possible to access the complete procedure call from inside the program?

Example of calling the program "fringes":

```
GDL> fringes, filename='testfilename.txt', number=500, width=300,99
```

Then inside the program:
```
pro fringes,samplesize,filename=filename,number=number,width=width

calling_sequence=???
openw,1,"calling_sequence"
printf,1,calling_sequence
close,1
...
end

and finally:

```
> cat calling_sequence
> fringes, filename='testfilename.txt', number=500, width=300,99
```

Can this be achieved?

**Comment by GillesDuvert** [comment]
_EXTRA may be what you need, see keyword inheritance in IDL documentation:
given:
```
pro fringes,_EXTRA = ex
  print, ex, /implied
  stop
end
```
you get
```
GDL> fringes, filename='testfilename.txt', number=500, width=300
{
    "WIDTH": 300,
    "NUMBER": 500,
    "FILENAME": "testfilename.txt"
}
```
but note that _EXTRA works only on Keywords, that why I did not add your example's parameter 99 in the call, The number of actual parameters in the calling list can be found by using the system function N_PARAMS, and if they are not 'passed' in the command, then their N_ELEMENTS() are 0.

combining both you could get your result I think

**Reply by GillesDuvert** [reply]
SCOPE_VARFTECH and SCOPE_VARNAME may be useful too, but IMHO this kind of instrospection is not really healthy in IDL.
